### DISCUSSION TITLE
Association does not exist, an issue with file structure?

### DISCUSSION THREAD

**Discussion by jmc-bbk** [sequelize/sequelize#15582] [Help & Questions] [answered]
I'm building an app with Express and Node, and using Sequelize to handle connections to a Postgres DB.

I have inconsistent behaviour on associations where even though an association is defined, Sequelize returns an error `SequelizeEagerLoadingError: user is not associated to Guest!`.

My basic application structure is as follows:

```
controllers
- users.js
- events.js
- guests.js
models
- user.js
- event.js
- guest.js
index.js
app.js
database.js
```

Where `User` has a M-M relationship with `Event` through `Guest`.

`database.js` handles connection/sequelize.authenticate()/sequelize.sync() and this is imported to each of the model files.

The relationship is defined in `Guest.js` as follows:

```
User.belongsToMany(Event, {
  through: Guest,
  foreignKey: 'userId'
})

Event.belongsToMany(User, {
  through: Guest,
  foreignKey: 'eventId'
})
```

However, when I try the following:

```
  const users = await Guest.findOne({
    where: {
      eventId:eventId
    },
    include: User
  })
```

I get the error message `SequelizeEagerLoadingError: user is not associated to Guest!`.

I'm convinced this must be an issue with file structure - as the Guest table is created appropriately in my Postgres DB - so Sequelize _must_ understand the association at some point.

Do I need to load the associations elsewhere? e.g. `app.js` `index.js`?

Appreciate any help here - as I cannot find any answers online.

**Comment by ephys** [comment, marked as answer]
In Sequelize 6, `belongsToMany` does not register the association from the through table to the other two models.

This is a feature that was added in Sequelize 7, but in Sequelize 6 you have to declare these associations explicitly

This is the concept of the ["super many-to-many"](https://sequelize.org/docs/v7/advanced-association-concepts/advanced-many-to-many/#through-tables-versus-normal-tables-and-the-super-many-to-many-association).
